### PR TITLE
fix(datastore,backup): half-open GetByHour boundary; drop dead backup loop var

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Two small, independent correctness fixes found during datastore tech-debt review.

**1. `GetByHour` / `CountByHour` double-counted the exact hour boundary** (`internal/datastore/v2/repository/detection_impl.go`). Both delegated to `GetByDateRange`/`CountByDateRange`, which use an inclusive upper bound (`detected_at <= ?`), passing `hourStart+3600` as the end. A detection whose `detected_at` landed exactly on the next hour's start was counted in both adjacent hours. The hour methods now use dedicated half-open helpers (`detected_at >= ? AND detected_at < ?`). The shared `GetByDateRange`/`CountByDateRange` keep their inclusive semantics, which other callers (e.g. full-day ranges in `dual_write.go`, where the end is computed as `midnight + 24h - 1ns`) intentionally rely on.

**2. Dead loop variable in backup `performBackupSteps`** (`internal/backup/sources/sqlite.go`). `remaining := total` was initialized and used in `for remaining > 0`, but never decremented; the loop only exited via `backupConn.Step()` returning done/err. Replaced with `for {`, removed the now-unused `total` parameter, and updated the single caller (`validatedTotal` is still logged). Behavior-preserving: `validatePageCount` already guarantees a positive page count before the loop, and `Step()` would return done immediately on an empty backup anyway.

## Test Plan
- [x] New `TestGetByHour_HalfOpenBoundary` seeds detections at `+10`, `+3599`, and exactly `+3600`; asserts the current hour counts 2 and the next hour counts the boundary exactly once. Confirmed it fails on the old inclusive code (counts 3) and passes after the fix.
- [x] `go test -race ./internal/datastore/v2/repository/ ./internal/backup/...`
- [x] `golangci-lint run` clean on changed packages
- [x] full `go build ./...`

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved SQLite backup reliability by switching to an open-ended chunking approach that continues until completion.
  * Fixed hourly detection retrieval and counting to correctly handle time boundary conditions.

* **Tests**
  * Added test coverage for time boundary handling in hourly detection queries.
  * Added test to verify pagination results remain accurate alongside full count reporting.

### CODE REVIEW

File: internal/datastore/v2/repository/detection_impl.go [hunk 1]
```diff
@@ -575,10 +575,27 @@ func (r *detectionRepository) GetByDateRange(ctx context.Context, start, end int
...
+func (r *detectionRepository) listByHalfOpenRange(ctx context.Context, start, end int64, limit, offset int) ([]*entities.Detection, int64, error) {
+	var dets []*entities.Detection
+	var total int64
+
+	query := r.db.WithContext(ctx).Table(r.tableName()).
+		Where("detected_at >= ? AND detected_at < ?", start, end)
+
+	if err := query.Count(&total).Error; err != nil {
+		return nil, 0, err
+	}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/datastore/v2/repository/detection_impl.go:597]: Good catch. I verified the LIMIT/ORDER/OFFSET are actually still honored here (added TestGetByHour_PaginationHonored: 5 rows in an hour, Limit=2 returns exactly 2 with total=5), so this didn't manifest as a live bug for this query shape. But reusing a non-session query across the Count and Find finishers is fragile, and the Search method in this same file already clones with Session(&gorm.Session{}), so I applied that pattern to listByHalfOpenRange in 43a718141 to match the established idiom and remove the latent risk.
